cpp/src/rolling/rolling_detail.cuh

```diff
@@ -1062,6 +1062,72 @@ struct rolling_window_launcher {
     return gather_map;
   }
 
+  /**
+   * @brief Purge entries for null inputs from gather_map, and adjust offsets.
+   */
+  void purge_null_entries(column_view const& input,
+                          std::unique_ptr<column>& gather_map,
+                          std::unique_ptr<column>& offsets,
+                          rmm::cuda_stream_view stream,
+                          rmm::mr::device_memory_resource* mr)
+  {
+    auto input_device_view = column_device_view::create(input, stream);
+
+    auto input_row_is_null = [d_input = *input_device_view] __device__(auto i) {
+      return d_input.is_null_nocheck(i);
+    };
+    auto input_row_not_null = [d_input = *input_device_view] __device__(auto i) {
+      return d_input.is_valid_nocheck(i);
+    };
+
+    auto num_child_nulls = thrust::count_if(rmm::exec_policy(stream),
+                                            gather_map->view().template begin<size_type>(),
+                                            gather_map->view().template end<size_type>(),
+                                            input_row_is_null);
```

Comment:
This looks like a much less efficient version of `column::null_count`. Can't this all be replaced by `input.null_count()`?

Reply:
> much less efficient version of `column::null_count`.

Do we mean `column_view::null_count(begin, end)`? That's brilliant. Let me try switching to that.

Reply:
> Can't this all be replaced by `input.null_count()`?

For the record, `input.null_count()` is already checked in the calling function. The check at Line#1088 is to see if the `gather_map` chooses any null elements from `input`. It doesn't look like `column::null_count()` or `column_view::null_count(begin,end)` makes that available.

There might be value in using `segmented_count_set_bits()`. I'm checking now.

Reply:
Ah, I see, you're doing a fused gather + null_count. I see now.